packages/dashboard/src/pages/CreateMaskWallet/CreateWalletForm/index.tsx

```diff
@@ -0,0 +1,187 @@
+import { memo, useEffect, useMemo } from 'react'
+import { Box, formHelperTextClasses, Typography } from '@mui/material'
+import { makeStyles, MaskColorVar } from '@masknet/theme'
+import { z as zod } from 'zod'
+import { useForm, Controller } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useNavigate, useSearchParams } from 'react-router-dom'
+import { SetupFrameController } from '../../../components/CreateWalletFrame/index.js'
+import { DashboardRoutes } from '@masknet/shared-base'
+import { useDashboardI18N } from '../../../locales/index.js'
+import { useAsyncRetry } from 'react-use'
+import { PluginServices } from '../../../API.js'
+import PasswordField from '../../../components/PasswordField/index.js'
+import { PrimaryButton } from '../../../components/PrimaryButton/index.js'
+
+const useStyles = makeStyles()((theme) => ({
+    container: {
+        width: '100%',
+        display: 'flex',
+        justifyContent: 'center',
+        flexDirection: 'column',
+    },
+    title: {
+        fontSize: 30,
+        margin: '12px 0',
+        lineHeight: '120%',
+        color: theme.palette.maskColor.main,
+    },
+    form: {
+        marginTop: 24,
+        width: '90%',
+        maxWidth: 720,
+    },
+    input: {
+        width: '100%',
+        marginTop: 10,
+    },
+    textField: {
+        background: theme.palette.mode === 'dark' ? '#1D2023' : '#F7F9FA',
+        padding: theme.spacing(1),
+        fontSize: 12,
+        lineHeight: '16px',
+        borderRadius: 6,
+        [`&.${formHelperTextClasses.error}`]: {
+            boxShadow: `0 0 0 ${theme.spacing(0.5)} ${MaskColorVar.redMain.alpha(0.2)}`,
+            border: `1px solid ${MaskColorVar.redMain.alpha(0.8)}`,
+        },
+    },
+    tips: {
+        fontSize: 14,
+        lineHeight: '18px',
+        fontFamily: 'Helvetica',
+        color: theme.palette.maskColor.second,
+    },
+    tipsBottom: {
+        fontSize: 14,
+        lineHeight: '18px',
+        fontFamily: 'Helvetica',
+        marginTop: 8,
+        marginBottom: 24,
+        color: theme.palette.maskColor.main,
+    },
+
+    second: {
+        fontSize: 14,
+        lineHeight: '18px',
+        color: theme.palette.maskColor.second,
+    },
+    helveticaBold: {
+        fontWeight: 700,
+        fontFamily: 'Helvetica',
+    },
+}))
+
+const CreateWalletForm = memo(() => {
+    const t = useDashboardI18N()
+    const { classes, cx } = useStyles()
+    const navigate = useNavigate()
+    const [searchParams] = useSearchParams()
+    const { value: hasPassword, loading, retry } = useAsyncRetry(PluginServices.Wallet.hasPassword, [])
+
+    useEffect(() => {
+        if (hasPassword) navigate(DashboardRoutes.CreateMaskWalletMnemonic)
```

Comment:
This will cause a flash of blank page.

Reply:
Removed, since Popup will not navigate to this page if you have password.